File: rs2_dogadjaj_Ba/dogadjaj_ba/dogadjaj_ba.Model/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace dogadjaj_ba.Model
{
    public class Event
    {
        public int EventId { get; set; }

        public string EventName { get; set; } = null!;

        public int? LokacijaId { get; set; }

        public DateTime EventDate { get; set; }

        public string? Opis { get; set; }

        public int? EventTypeId { get; set; }
        public string? StateMachine { get; set; }
        public byte[]? EventImage { get; set; }
    }
}

File: mobile/lib/models/event.dart
class Event {
  final int? eventId;
  final String? eventName;
  final int? lokacijaId; 
  final DateTime? eventDate;
  final int? eventTypeId;
  final int? subType;
  final String? opis;
  final String? stateMachine;
  final String? eventImage;

  Event({
    this.eventId,
    this.eventName,
    this.lokacijaId, 
    this.eventDate,
    this.eventTypeId,
    this.subType,
    this.opis,
    this.stateMachine,
    this.eventImage,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      eventId: json['eventId'],
      eventName: json['eventName'] ?? "Nepoznat događaj",
      lokacijaId: json['lokacijaId'], 
      eventDate: json['eventDate'] != null ? DateTime.parse(json['eventDate']) : null,
      eventTypeId: json['eventType'],
      subType: json['subType'],
      opis: json['opis'],
      stateMachine: json['stateMachine'],
      eventImage: json['eventImage'],
    );
  }



  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['eventId'] = eventId;
    data['eventName'] = eventName;
    data['eventTypeId'] = eventTypeId;
    data['opis'] = opis;
 data['lokacijaId'] = lokacijaId;
    return data;
  }

}